src/krux/input.py

```diff
@@ -42,35 +49,79 @@ def __init__(self):
         self.enter = GPIO(GPIO.GPIOHS21, GPIO.IN, GPIO.PULL_UP)
         fm.register(board.config["krux.pins"]["BUTTON_B"], fm.fpioa.GPIOHS22)
         self.page = GPIO(GPIO.GPIOHS22, GPIO.IN, GPIO.PULL_UP)
+        if "BUTTON_C" in board.config["krux.pins"]:
+            fm.register(board.config["krux.pins"]["BUTTON_C"], fm.fpioa.GPIOHS0)
+            self.previous = GPIO(GPIO.GPIOHS0, GPIO.IN, GPIO.PULL_UP)
+        self.has_touch = board.config["lcd"]["touch"]
+        self.touch = (
+            Touch(board.config["lcd"]["width"], board.config["lcd"]["height"], 50)
+            if self.has_touch
+            else None
+        )
+
+    def get_button_c(self):
+        """Intermediary method to pull button C state, if available"""
+        if "BUTTON_C" in board.config["krux.pins"]:
+            return self.previous.value()
+        return RELEASED
+
+    def get_touch_state(self):
+        """Intermediary method to pull touch state, if touch available"""
+        if self.has_touch:
+            return self.touch.get_state()
+        return TOUCH_IDLE
+
+    def wait_for_release(self):
+        """Loop until all buttons are released (if currently pressed)"""
+        while (
+            self.enter.value() == PRESSED
+            or self.page.value() == PRESSED
+            or self.get_button_c() == PRESSED
+            or self.get_touch_state() == TOUCH_PRESSED
+        ):
+            self.entropy += 1
+
+    def wait_for_press(self, block=True):
+        """Wait for first button press"""
+        time_frame = time.ticks_ms()
+        while (
+            self.enter.value() == RELEASED
+            and self.page.value() == RELEASED
+            and self.get_button_c() == RELEASED
+            and self.get_touch_state() == TOUCH_IDLE
+        ):
+            wdt.feed() #here is were krux spend most of time
+            if not block and time.ticks_ms() > time_frame + QR_ANIM_PERIOD:
+                break
 
     def wait_for_button(self, block=True):
         """Waits for any button to release, optionally blocking if block=True.
         Returns the button that was released, or None if nonblocking.
         """
-        # Loop until all buttons are released (if currently pressed)
-        while self.enter.value() == PRESSED or self.page.value() == PRESSED:
-            wdt.feed()
-            self.entropy += 1
-
-        # Wait for first button press
-        checks = 0
-        while self.enter.value() == RELEASED and self.page.value() == RELEASED:
-            wdt.feed()
-            checks += 1
-            if not block and checks > NONBLOCKING_CHECKS:
-                break
+        self.wait_for_release()
+        self.wait_for_press(block)
 
         if self.enter.value() == PRESSED:
             # Wait for release
             while self.enter.value() == PRESSED:
```

Comment:
I think the `wdt.feed` should remain in these while loops, otherwise if a user held down a button press for > 30 seconds, the device would reset